Create nft.ts

yahyamohmuedpro99 opened the pull request:
Add the nft mapping handler with these messages:

- nft issue
- mint
- burn
- freeze

yahyamohmuedpro99 committed:
Create nft.ts

add the nft mapping handler with this massages 
-nft issue
-mint
-burn
-freeze

metmirr commented on subqueries/coreum-testnet/src/mappings/nft.ts:
```diff
@@ -0,0 +1,74 @@
+export async function handleNftMsgMint(message:CosmosMessage<MsgMintMessageNft>):Promise<void>{
+    // const mint=message.msg.decodedMsg.data
+
+    const transaction = createTransaction('MsgMint',message)
```
You need to rename `MsgMint` here as well (maybe you can change it to `MsgMintNFT`. It will create a conflict on the Kafka consumer side.

yahyamohmuedpro99 replied:
can you check now please

metmirr commented on subqueries/coreum-testnet/src/mappings/nft.ts:
```diff
@@ -0,0 +1,74 @@
+    uriHash: string;
+    data: any;
+}
+export async function handleNftMsgMint(message:CosmosMessage<MsgMintMessageNft>):Promise<void>{
```
The generic type should be the interface above. Don't import types from `../types/CosmosMessageTypes.ts` because you define your own interfaces here.

yahyamohmuedpro99 replied:
can you check now please

metmirr commented on subqueries/coreum-testnet/src/mappings/nft.ts:
```diff
@@ -0,0 +1,74 @@
+import { CosmosMessage } from '@subql/types-cosmos'
+import {MsgIssueMessage,MsgMintMessageNft,MsgBurnMessageNft,MsgFreezeMessageNft} from '../types/CosmosMessageTypes'
```
No need for this line anymore!

yahyamohmuedpro99 committed:
Update nft.ts

update the types of generic to be from the interface 
rename the name of transactions of nft to not make conflict in kafka

metmirr reviewed (approved):
Looks great!
